src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -459,9 +459,7 @@ function assignOptionIds(
     (
       option: OptionDescriptor | ActionListItemDescriptor,
       optionIndex: number,
-    ) => {
-      option.id = `${comboBoxId}-${optionIndex}`;
-    },
+    ) => Object.assign({id: `${comboBoxId}-${optionIndex}`}, option),
```

Comment:
```suggestion
) => ({id: `${comboBoxId}-${optionIndex}`, ...option}),
```

A preference we have in our linter. I'm surprised it didn't auto-correct for you on save.